src/Stack/Setup.hs

```diff
@@ -140,6 +140,8 @@ data SetupOpts = SetupOpts
     , soptsUpgradeCabal :: !Bool
     -- ^ Upgrade the global Cabal library in the database to the newest
     -- version. Only works reliably with a stack-managed installation.
+    , soptsInstallCabal :: !(Maybe Version)
+    -- ^ Install a specific version of Cabal.
```

Comment:
Oh, I see that [options with optional arguments are not supported](https://github.com/pcapriotti/optparse-applicative#faq).